src/dftbp/wavegrid/molorb/cuda/device_params.cuh

```diff
@@ -0,0 +1,347 @@
+/*-------------------------------------------------------------------------------------------------*
+ *  DFTB+: general package for performing fast atomistic simulations                               *
+ *  Copyright (C) 2006 - 2025  DFTB+ developers group                                              *
+ *                                                                                                 *
+ *  See the LICENSE file for terms of usage and distribution.                                      *
+ *-------------------------------------------------------------------------------------------------*/
+// This file contains the kernel parameter struct and
+// helpers to manage allocating and copying data to the GPU.
+#pragma once
+
+#include <memory>
+#include <stdexcept>
+#include <vector>
+
+#include <cuda_runtime.h>
+#include <thrust/complex.h>
+
+#include "kernel.cuh"
+#include "utils.cuh"
+
+
+using complexd = thrust::complex<double>;
+
+
+/*
+ * A simple RAII wrapper for device memory.
+ * Use get() to retrieve the raw pointer.
+ */
+template <typename T>
+class DeviceBuffer {
+   public:
+    // Allocate nothing
+    DeviceBuffer() = default;
+
+    // Allocate only
+    explicit DeviceBuffer(size_t count) { allocate(count); }
+
+    // Allocate and copy from host
+    DeviceBuffer(const T* host_ptr, size_t count) {
+        allocate(count);
+        copy_to_device(host_ptr, count);
+    }
+
+    // Destructor frees memory on device
+    ~DeviceBuffer() { deallocate(); }
+
+    // Disable copy constructor, assignment
+    DeviceBuffer(const DeviceBuffer&)            = delete;
+    DeviceBuffer& operator=(const DeviceBuffer&) = delete;
+
+    // Enable move semantics
+    DeviceBuffer(DeviceBuffer&& other) noexcept :
+        _devicePtr(other._devicePtr),
+        _count(other._count) {
+        other._devicePtr = nullptr;
+        other._count     = 0;
+    }
+    DeviceBuffer& operator=(DeviceBuffer&& other) noexcept {
+        if (this != &other) {
+            deallocate();
+            _devicePtr       = other._devicePtr;
+            _count           = other._count;
+            other._devicePtr = nullptr;
+            other._count     = 0;
+        }
+        return *this;
```

Comment:
Do we ever need / use those contructors? If not, we should probably simplify the code and leave them away.